src/WebJobs.Script.WebHost/Diagnostics/WebHostMetricsLoggerDecorator.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public class WebHostMetricsLoggerDecorator : IMetricsLogger
```

Comment:
Let's make this internal. Also add a comment as to why we're doing this -- i.e. we don't want it to dispose but DryIoc will do it anyway (and it'd be a break).

Reply:
Let's also not use the pattern name in the type name. Something like `NonDisposableMetricsLogger` (or a better name if  you wish) would be more appropriate.